src/main/kotlin/com/zepben/evolve/streaming/grpc/Connect.kt

```diff
@@ -0,0 +1,229 @@
+    ): GrpcChannel =
+        GrpcChannelBuilder().forAddress(host, rpcPort).build()
+
+    /**
+     * Create a `GrpcChannel` that communicates with the gRPC service using SSL/TLS transport security.
+     *
+     * @param host The hostname where the gRPC service is hosted
+     * @param rpcPort The port of the gRPC service
+     * @param caFilename The filename of a truststore containing additional trusted root certificates. This parameter is optional
+     *                   and defaults to null, in which case only the system CAs are used to verify certificates.
+     * @return An encrypted connection to the gRPC service
+     */
+    @JvmStatic
+    @JvmOverloads
+    fun connectTls(
+        host: String = "localhost",
+        rpcPort: Int = 50051,
+        caFilename: String? = null
+    ): GrpcChannel =
+        GrpcChannelBuilder().forAddress(host, rpcPort).makeSecure(rootCertificates = caFilename).build()
+
+    /**
+     * Create a `GrpcChannel` that communicates with the gRPC service using SSL/TLS transport security and the OAuth client credentials flow.
+     * The OAuth provider's domain and the "audience" parameter of the token request are fetched as JSON from a specified URL.
+     *
+     * @param clientId The client ID of the OAuth application to authenticate for
+     * @param clientSecret The client secret of the OAuth application to authenticate for
+     * @param confAddress The address of the authentication configuration
+     * @param confCAFilename The filename of a truststore containing additional trusted root certificates for fetching the authentication configuration.
+     *                       This parameter is optional and defaults to null, in which case only the system CAs are used to verify certificates.
+     * @param authCAFilename The filename of a truststore containing additional trusted root certificates for fetching the OAuth tokens.
+     *                       This parameter is optional and defaults to null, in which case only the system CAs are used to verify certificates.
+     * @param host The hostname where the gRPC service is hosted
+     * @param rpcPort The port of the gRPC service
+     * @param caFilename The filename of a truststore containing additional trusted root certificates. This parameter is optional
+     *                   and defaults to null, in which case only the system CAs are used to verify certificates.
+     * @return An Auth0-authenticated, encrypted connection to the gRPC service. If the authentication configuration specifies that no authentication is
+     *         required, a non-authenticated, encrypted connection is returned instead.
+     */
+    @JvmStatic
+    @JvmOverloads
+    fun connectWithSecret(
+        clientId: String,
+        clientSecret: String,
+        confAddress: String? = null,
+        confCAFilename: String? = null,
+        authCAFilename: String? = null,
+        host: String = "localhost",
+        rpcPort: Int = 50051,
```

Comment:
is there any reason you put host and rpcPort near the end? I'd prefer them up before confAddress as they're very likely to be changed from the defaults.

Reply:
I just did that to be consistent with the python version. I can see the argument for higher billing now